lib/index.ts

```diff
@@ -0,0 +1,561 @@
+/*
+Copyright 2017 Resin.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+/* tslint:disable: max-classes-per-file */
+
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as querystring from 'querystring';
+import * as request from 'request-promise';
+import TypedError = require('typed-error');
```

Comment:
I've defined it in the way it requires to export the module as a class type.
Doing it such as:
```
import { TypedError} from 'typed-error';
```
Will actually transpile to:
```
const TypedError = require("typed-error").TypedError;
```
Which will fail as there is no `TypedError` property on the module. The same is true of doing a default export. Unfortunately, I have no control over the TypedError interface.